### ISSUE TITLE
Duplicate response headers overwritten, not appended

### ISSUE THREAD

**Issue by sandydoo** [ember-fastboot/fastboot-express-middleware#33]
FastBoot provides us with an API to `append` headers to the response. If you set the same header more than once, FastBoot will store the values in an array. However, the Express middleware then loops through the flattened array of header entries and sets them using `res.set()`, which overwrites any duplicate headers with the last value.

https://github.com/ember-fastboot/fastboot-express-middleware/blob/5a0981d4394bd68acda70ea9e634ec8c2e05c513/src/index.js#L37-L42

### Example

If you wanted to return several cookies, you might try the following:
```javascript
let responseHeaders = this.get('fastboot.response.headers');
responseHeaders.append('set-cookie', 'first_cookie=...; path=/');
responseHeaders.append('set-cookie', 'second_cookie=...; path=/');
```
The first cookie gets overwritten by the second in the response.

### In the wild

Both `ember-simple-auth`  and `ember-cookies` seem to expect the cookies to be appended, not overwritten with the last value. 

In `ember-cookies`: https://github.com/simplabs/ember-cookies/blob/0a0fb0e4d85b70041258829c5e7e2e4f1359ca7b/addon/services/cookies.js#L94-L105
```javascript
  _writeFastBootCookie(name, value, options = {}) {
    let responseHeaders = this.get('_fastBoot.response.headers');
    let serializedCookie = this._serializeCookie(...arguments);


    if (!isEmpty(options.maxAge)) {
      options.maxAge *= 1000;
    }


    this._cacheFastBootCookie(...arguments);


    responseHeaders.append('set-cookie', serializedCookie);
  },
```

This innocuous code then produces subtle bugs in code like this in `ember-simple-auth`: https://github.com/simplabs/ember-simple-auth/blob/master/addon/session-stores/cookie.js#L246
```javascript
this.get('_cookies').write(this.get('cookieName'), value, cookieOptions);
if (!isEmpty(expiration)) {
  let expirationCookieName = `${this.get('cookieName')}-expiration_time`;
  let cachedExpirationTime = this.get('_cookies').read(expirationCookieName);
  this.get('_cookies').write(expirationCookieName, this.get('cookieExpirationTime') || cachedExpirationTime, cookieOptions);
}
```
`ember-simple-auth` depends on the cookie store in FastBoot, so if you set an expiration date on your auth cookies, this expiration cookie will overwrite your auth cookie.

I assume this is not really the intended behaviour, especially for the `append` method of the fetch header API. I know Express previously provided all sorts of patches, including automatically appending duplicate header values (https://github.com/expressjs/express/wiki/Migrating-from-3.x-to-4.x#ressetheaderset-cookie-val).

The best solution I can come up with would be to handle the cookies with `res.cookie` here and let users pass in cookies via the FastBoot service. That's gonna require changes in 3 repos at least, so not sure how people feel about that.

We can filter out the cookie headers and set them all at once (which correctly sets multiple `set-cookie` headers), but we don't want to send more than one cookie with the same name. `ember-simple-auth` in particular will write the auth cookie 3-4 times for some reason. So, we'd have to parse the serialized cookies to check for duplicates – doesn't sound like a great idea.

**Comment by marcoow** [comment]
Right, that should probably just use `res.cookie` [here](https://github.com/ember-fastboot/fastboot-express-middleware/blob/5a0981d4394bd68acda70ea9e634ec8c2e05c513/src/index.js#L41).

**Comment by bobisjan** [comment]
I've encountered this issue recently, I see that #36 fixes this for `Set-Cookie` header, but all the "multivalue" headers should be supported?

Since both [ember-cli](https://github.com/ember-cli/ember-cli/blob/master/package.json#L82) and [fastboot-app-server](https://github.com/ember-fastboot/fastboot-app-server/blob/master/package.json#L31) requires `express#4.11` then the [response.append()](https://expressjs.com/en/4x/api.html#res.append) could be used to fix the issue?

I'm able to provide a PR with the fix, if using `response.append()` would be appropriate. /cc @rwjblue

**Comment by rwjblue** [comment]
@bobisjan ya that seems good

**Comment by sandydoo** [comment]
Fixed in #60.
